rust/agama-lib/share/storage.schema.json

```diff
@@ -63,6 +68,89 @@
         }
       }
     },
+    "mdRaidElement": {
+      "anyOf": [
+        { "$ref": "#/$defs/formattedMdRaid" },
+        { "$ref": "#/$defs/partitionedMdRaid" }
+      ]
+    },
+    "formattedMdRaid": {
+      "description": "MD RAID without a partition table (directly formatted).",
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["level", "filesystem"],
+      "properties": {
+        "alias": { "$ref": "#/$defs/alias" },
+        "name": { "$ref": "#/$defs/mdRaidName" },
+        "level": { "$ref": "#/$defs/mdRaidLevel" },
+        "parity": { "$ref": "#/$defs/mdRaidParity" },
+        "chunkSize": { "$ref": "#/$defs/sizeValue" },
+        "devices": { "$ref": "#/$defs/mdRaidDevices" },
+        "encryption": { "$ref": "#/$defs/encryption" },
+        "filesystem": { "$ref": "#/$defs/filesystem" }
+      }
+    },
+    "partitionedMdRaid": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["level"],
+      "properties": {
+        "alias": { "$ref": "#/$defs/alias" },
+        "name": { "$ref": "#/$defs/mdRaidName" },
+        "level": { "$ref": "#/$defs/mdRaidLevel" },
+        "parity": { "$ref": "#/$defs/mdRaidParity" },
+        "chunkSize": { "$ref": "#/$defs/sizeValue" },
+        "devices": { "$ref": "#/$defs/mdRaidDevices" },
+        "ptableType": { "$ref": "#/$defs/ptableType" },
+        "partitions": {
+          "type": "array",
+          "items": { "$ref": "#/$defs/partitionElement" }
+        }
+      }
+    },
+    "mdRaidName": {
+      "description": "MD base name.",
+      "type": "string",
+      "examples": ["system"]
+    },
+    "mdRaidLevel": {
+      "title": "MD level",
+      "enum": [
+        "raid0",
+        "raid1",
+        "raid5",
+        "raid6",
+        "raid10"
+      ]
+    },
+    "mdRaidParity": {
+      "title": "MD parity",
+      "description": "Only applies to raid5, raid6 and raid10",
+      "enum": [
+        "left_asymmetric",
+        "left_symmetric",
+        "right_asymmetric",
+        "right_symmetric",
+        "first",
+        "last",
+        "left_asymmetric_6",
```

Comment:
Not sure whether we should support the variants with `_6` for creating new RAIDs.